### PR TITLE
Feature/copernicus swh

### LINKED ISSUE #637: L3 SWH from Copernicus
## Description
Convert the satellite-based along-track significant wave height product [L3 SWH from Copernicus](https://resources.marine.copernicus.eu/product-detail/WAVE_GLO_WAV_L3_SWH_NRT_OBSERVATIONS_014_001/INFORMATION) to ioda format. This data set is intended for wave data assimilation.

### PR DESCRIPTION
**Pull request by clouden90**
## Description
This feature branch contains a ioda converter for the level 3 satellite-based along-track significant wave height (swh) from the Copernicus. The converter was based on @guillaumevernieres's work on L4 Copernicus ADT and outputs directly to ioda v2 format. 

Added feature:
- 1 converter (based on ./src/marine/copernicus_l4adt2ioda.py );
- 1 subsampled swh file in netcdf format from Copernicus;
- A reference output that was separately tested by jeda-soca wave da interface.

### Issue(s) addressed
- fixes #637 

## Acceptance Criteria (Definition of Done)
Converts the near real-time L3 SWH files to ioda format.

## Dependencies

- [x] waiting on jcsda-internal/ioda/pull/499

### CODE REVIEW

File: src/marine/copernicus_l3swh2ioda.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    'swh': ['nlocs'],
+}
+
+
+class copernicus(object):
+    def __init__(self, filename, factor):
+        # Create a simple data structure for Copernicus L3 swh near-realtime product obtained from
+        # ftp://nrt.cmems-du.eu/Core/WAVE_GLO_WAV_L3_SWH_NRT_OBSERVATIONS_014_001/
+        ncd = nc.Dataset(filename, 'r')
+        self.lons = np.squeeze(ncd.variables['longitude'][:]).ravel()
```
Comment (guillaumevernieres) [inline src/marine/copernicus_l3swh2ioda.py:56]: No need for _squeezing_ and your arrays are already 1D, so no need for _ravel_ . That's true for all the lines below making use of these 2 methods.

File: src/marine/copernicus_l3swh2ioda.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        self.date = ncd.getncattr('first_meas_time')
+        ncd.close()
+
+        # set time stamp for all obs
+        self.datetime = np.empty_like(self.swh, dtype=object)
+        for t in range(len(self.datetime)):
+            self.datetime[t] = datetime.utcfromtimestamp(self.time[t]+datetime(2000, 1, 1, 0, tzinfo=pytz.UTC).timestamp()).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        # Remove observations out of sane bounds
+        qci = np.where(self.swh > 0.001)
```
Comment (guillaumevernieres) [inline src/marine/copernicus_l3swh2ioda.py:70]: dumb questions, why are _swh_ obs between 0.0 and 0.001 bad?

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Thanks for doing that work @clouden90 . 
I have a few minor comments.

File: src/marine/copernicus_l3swh2ioda.py (hunk 1 above)
Comment (clouden90) [reply src/marine/copernicus_l3swh2ioda.py:56]: Thx for point this out, @guillaumevernieres ! I have removed these methods.

File: src/marine/copernicus_l3swh2ioda.py (hunk 2 above)
Comment (clouden90) [reply src/marine/copernicus_l3swh2ioda.py:70]: Good catch, @guillaumevernieres. This is just an arbitrary value I picked. Actually we should consider all swh obs > 0.0, especially these data has been pre-QC/validated against  wave buoy data. Changed this line to self.swh > 0.0.

**Pull request merged**